### ISSUE TITLE
faster bulk deletes in safekeeper

### ISSUE THREAD

**Issue by arpad-m** [neondatabase/neon#9152]
The safekeepers right now delete with a speed of around 70k files per minute. However, sometimes there is projects with millions of files (>50 TB of WAL). We should support a mode where if the number of files to delete is larger than 10k, we issue deletion requests in parallel.

cc slack thread: https://neondb.slack.com/archives/C03H1K0PGKH/p1727278511234979

**Comment by jcsp** [comment]
Closing in favor of https://github.com/neondatabase/neon/issues/10265
